llvm/lib/Support/GlobPattern.cpp

```diff
@@ -143,6 +143,15 @@ GlobPattern::create(StringRef S, std::optional<size_t> MaxSubPatterns) {
     return Pat;
   S = S.substr(PrefixSize);
 
+  // Just in case we stop on unmatched opening brackets.
+  size_t SuffixStart = S.find_last_of("?*[]{}\\");
+  assert(SuffixStart != std::string::npos);
+  if (S[SuffixStart] == '\\')
```

Comment:
Can't a `\` escape be multiple characters, e.g. for a oct escape?

Reply:
It does not look like, see line 239